### ISSUE TITLE
max-elements in leaflist and list not work when input data is  in JSON format

### ISSUE THREAD

**Issue by KC-Su** [CESNET/libyang#267]
Hi, seems JSON parser has problem in `devel`(0.12.77).

Here is the schema(enabling the `feature_foo` in both cases):
```yang
module schema {

  yang-version 1.1;
  namespace "http://example.com/schema";
 
  prefix 'test';

  feature feature_foo;
  identity ident_foo;
  typedef tpdf_foo {
    type int32;
  }

  leaf-list leaflist_test {
    when "2>1";
    if-feature "feature_foo";
    type string;
    units "unit";
    must "2>1";
    config true;
    min-elements 0;
    max-elements 2;
    ordered-by user;
    status current;
    description "my description";
    reference "my reference";
  }
}
```

when I input data in XML format, it works with log `libyang[0]: Too many "leaflist_test" elements.`
```xml
<leaflist_test xmlns="http://example.com/schema">first string</leaflist_test>
<leaflist_test xmlns="http://example.com/schema">second string</leaflist_test>
<leaflist_test xmlns="http://example.com/schema">third string</leaflist_test>
```
but strangely get data node atfer calling `lyd_parse_path`  if input data format is json
```json
{
	"schema:leaflist_test": [
		"first string",
		"second string",
		"third string"
	]
}
```

And similar issue occur when using `list`.

**Comment by rkrejci** [comment]
Can you share how you load/validate the data? Because I see:
```
$ ./yanglint --version
yanglint 0.12.77
$ ./yanglint schema.yang -F schema:* schema.json
err : Too many "leaflist_test" elements. (/schema:leaflist_test[.='third string'])
```

**Comment by KC-Su** [comment]
Hi,

No problem,
Here is the partial C code with using [CUnit](http://cunit.sourceforge.net/):
```c
void __tear_down(struct ly_ctx *context, struct lyd_node *instance, struct lyd_node *reply_instance)
{
	// free data instance first if existing
	if (instance != NULL) {
		lyd_free_withsiblings(instance);
	}

	// free reply data instance if existing
	if (reply_instance != NULL) {
		lyd_free_withsiblings(reply_instance);
	}

	// free context (including shema tree) last if existing
	if (context != NULL) {
		ly_ctx_destroy(context, NULL);
	}
	return;
}
int main() {
	struct ly_ctx *context = NULL;
	const struct lys_module *module = NULL;
	struct lyd_node *data = NULL;
	struct lyd_node_leaf_list *leaf_data = NULL;

	context = ly_ctx_new("./");
	CU_ASSERT_FALSE_FATAL(context == NULL);
	// loading yang module
	module = ly_ctx_load_module(context, "module", NULL);
	CU_ASSERT_FALSE_FATAL(module == NULL);
       //enable feature_foo
	CU_ASSERT_FALSE_FATAL(lys_features_enable(module, "feature_foo"));

	data = lyd_parse_path(context, "./data.json", LYD_JSON, LYD_OPT_CONFIG);
	CU_ASSERT_FALSE_FATAL(data != NULL);

	__tear_down(context, data, NULL);
	return 1;
}
```

and the gdb information:
```
1921		data = lyd_parse_path(context, "./data.json", LYD_JSON, LYD_OPT_CONFIG);
(gdb) n
1922		CU_ASSERT_FALSE_FATAL(data != NULL);
(gdb) p data
$1 = (struct lyd_node *) 0x6f8f00
(gdb) p *data
$2 = {schema = 0x705910, validity = 0 '\000', dflt = 0 '\000', when_status = 6 '\006', attr = 0x0, next = 0x0, prev = 0x7060a0, parent = 0x0, child = 0x6f9fb0}
(gdb) p *data->schema
$3 = {name = 0x709d80 "leaflist_test", dsc = 0x709f70 "my description", ref = 0x709ff0 "my reference", flags = 269, nacm = 0, module = 0x701420, nodetype = LYS_LEAFLIST, parent = 0x0, child = 0x0, next = 0x0, 
  prev = 0x705910, priv = 0x0, iffeature = 0x709ec0, iffeature_size = 1 '\001'}
```

And I get correct results while run with previous version (0.12.67).
Thanks

Best.

**Comment by rkrejci** [comment]
Thanks, please try 0.12.78.

**Comment by KC-Su** [comment]
It's fixed in 0.12.78. Thanks!
